realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -261,33 +282,23 @@ public static void validateTable(ImplicitTransaction transaction) {
             if (!transaction.hasTable("class_AllTypes")) {
                 throw new IllegalStateException("Missing table 'class_AllTypes' for column 'columnObject'");
             }
-            if(!columnTypes.containsKey("columnRealmList")) {
+            Table table_7 = transaction.getTable("class_AllTypes");
+            if (!table.getLinkTarget(INDEX_COLUMNOBJECT).equals(table_7)) {
```

Comment:
Isn't it dangerous to test for type validity before checking if the field exists at all?